### DISCUSSION TITLE
Adjust supported OS installation methods on macOS

### DISCUSSION THREAD

**Discussion by agners** [home-assistant/architecture#1118] [General]
# Context

Today, the official documentation includes VirtualBox and Core installation. Both are supported according to [ADR-0015](https://github.com/home-assistant/architecture/blob/master/adr/0015-home-assistant-os.md) and [ADR-0016](https://github.com/home-assistant/architecture/blob/master/adr/0016-home-assistant-core.md). ADR-0015 does not explicitly state macOS for VirtualBox, but it also does not explicitly mention other operating systems as Hypervisor hosts.

A rather common request is users trying to get Home Assistant running on newer macOS arm64 based notebooks. VirtualBox today does not support macOS on `arm64` (aka `aarch64`). There are preview builds available currently.

The [macOS installation instructions](https://www.home-assistant.io/installation/macos) today have a hint that UTM can be used. UTM has grown quite some popularity over VirtualBox nowadays. It is QEMU based and can readily consume qcow2.

This discussion is only concerned about the Operating System installation method.

While macOS is typically not well suited for a server, it is still useful to have documentation and officially support macOS for testing, toe dipping and potentially also a server installation on a Apple Mac Mini (which nowadays also use the arm64 architecture).

# Proposal

Declare UTM as official and only supported Hypervisor under macOS. Supported are both x86_64 (through `ova` OS image) and arm64 (through `generic-aarch64` OS image). 

Adjust documentation accordingly (remove VirtualBox documentation from the macOS page, and document UTM).

**Comment by frenck** [comment]
> While macOS is typically not well suited for a server, it is still useful to have documentation and officially support macOS for testing,

☝️ Just that, for me personally, makes it that I think we should support it officially at all. 

I would argue we should remove support for VirtualBox and not add UTM for the reason you mentioned: It is nice for testing, not actual use.

**Reply by agners** [reply]
> ☝️ Just that, for me personally, makes it that I think we should support it officially at all.

I guess there is a "not" missing?

> I would argue we should remove support for VirtualBox and not add UTM for the reason you mentioned: It is nice for testing, not actual use.

I disagree.

That would mean we essentially drop support, for any sensible way to run on macOS.

But users will still install it. If you made up your mind to run HA on a macOS system, people will do it :man_shrugging: 

One of our core values is choice. This fundamentally takes choice away, which goes against our values.

Sure, there is always a tradeoff: Maintenance vs. how much impact such an additional choice actually have. Maybe macOS doesn't have that many users, but the expected maintenance is also rather low here: UTM is QEMU based, which we support anyways. The main effort here is documenting it once. Then we should be largely fine :man_shrugging: .

Having "choice" as core value and not officially supporting one of the major OS'es just feels wrong :man_shrugging:

**Comment by sairon** [comment]
I tend to agree with @frenck, [ADR-0012](https://github.com/home-assistant/architecture/blob/master/adr/0012-define-supported-installation-method.md) says:

> Supported means the installation method is tested and documented in the official documentation. Running Home Assistant using such a supported method, **leads to the optimal user experience**.

Currently we don't do any testing (continuous or manual) of macOS installation methods, so I'm hesitant to give any guarantees. Just recently we got [this report](https://github.com/home-assistant/operating-system/issues/3472) of rather seriously broken macOS (aarch64) setup which kind of illustrates what is the current state.

So if were up to me, I wouldn't declare any official support in ADR for the reasons mentioned above. I don't have any strong opinion about the documentation, I think we can both keep VirtualBox and add UTM, but at the same time add a mention noting this is not a recommended/supported setup and it has limited testing.
